src/transformStateWithClones.js

```diff
@@ -65,7 +65,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const currentState = { ...state };
+  const result = [];
+
+  transforms.forEach(item => {
+    switch (item.operation) {
+      case 'addProperties':
+        Object.keys(item.properties).forEach(key => {
+          currentState[key] = item.properties[key];
+        });
+        break;
+      case 'removeProperties':
+        item.properties.forEach(key => delete currentState[key]);
+        break;
+      case 'clear':
+        Object.keys(currentState).forEach(key => delete currentState[key]);
```

Comment:
It's ok but you also may use 'for...in' loop to iterate through object keys and not creating a new array of them using 'Object.keys;
```        
for (const key in currentState) {
  delete currentState[key];
}
```

Reply:
Okay, I'll try it!